src/sagas/pushNotification.js

```diff
@@ -0,0 +1,440 @@
+      deviceId,
+      enablePush: true,
+    });
+
+    if (success) {
+      const enabledAt = Date.now();
+      STORE.setItem(pushNotificationKey.enabledAt, enabledAt);
+      STORE.setItem(pushNotificationKey.hasBeenEnabled, true);
+      yield put(pushRegisterSuccess({ enabled: true, hasBeenEnabled: true, enabledAt }));
+    } else {
+      // NOTE: theoretically, this should never happen because when the client call fails, it throws an error
+      yield put(pushRegisterFailed());
+    }
+  } catch (error) {
+    console.error('Error registering device for the first time: ', error.cause);
+    yield put(pushRegisterFailed());
+  }
+}
+
+/**
+ * This function is responsible for registering the device on the wallet-service in the event
+ * of renewing the registration.
+ */
+export function* registration({ payload: { enabled, showAmountEnabled, deviceId } }) {
+  yield put(pushLoadWalletRequested());
+
+  // wait for the wallet to be loaded
+  const [loadWalletSuccess, loadWalletFail] = yield race([
+    take(types.PUSH_WALLET_LOAD_SUCCESS),
+    take(types.PUSH_WALLET_LOAD_FAILED)
+  ]);
+
+  if (loadWalletFail) {
+    yield put(pushRegisterFailed());
+    return;
+  }
+
+  const { walletService } = loadWalletSuccess.payload;
+  try {
+    const { success } = yield call(pushLib.PushNotification.registerDevice, walletService, {
+      pushProvider: Platform.OS,
+      deviceId,
+      enablePush: true,
+    });
+
+    if (success) {
+      const enabledAt = Date.now();
+      STORE.setItem(pushNotificationKey.enabledAt, enabledAt);
+      yield put(pushRegisterSuccess({ enablePush: enabled, enableShowAmounts: showAmountEnabled, hasBeenEnabled: true, enabledAt }));
+    } else {
+      // NOTE: theoretically, this should never happen because when the client call fails, it throws an error
+      yield put(pushRegisterFailed());
+    }
+  } catch (error) {
+    console.error('Error registering device: ', error.cause);
+    yield put(pushRegisterFailed());
+  }
+}
+
+/**
+ * This function is responsible for updating the registration of the device on the wallet-service
+ * in the event of changing the settings of the push notifications.
+ * @param {{ payload: { enabled: boolean, showAmountEnabled: boolean, deviceId: string } }} action
+ */
+export function* updateRegistration({ payload: { enabled, showAmountEnabled, deviceId } }) {
+  yield put(pushLoadWalletRequested());
+
+  // wait for the wallet to be loaded
+  const [loadWalletSuccess, loadWalletFail] = yield race([
+    take(types.PUSH_WALLET_LOAD_SUCCESS),
+    take(types.PUSH_WALLET_LOAD_FAILED)
+  ]);
+
+  if (loadWalletFail) {
+    yield put(pushRegisterFailed());
+    return;
+  }
+
+  const { walletService } = loadWalletSuccess.payload;
+  try {
+    const { success } = yield call(pushLib.PushNotification.updateDevice, walletService, {
+      deviceId,
+      enablePush: enabled,
+      enableShowAmounts: showAmountEnabled,
+    });
+    if (success) {
```

Comment:
Don't you need to update `enabledAt` in the `STORE` to `0`, in case `enabled` is `false`?